File: Cpp/1796_Second_Largest_Digit_in_a_String.cpp
class Solution {
public:
    int secondHighest(string s) {
        int result = -1;
        int largest_num = -1;
        
        for(auto& ch : s) {
            if (ch < '0' || ch > '9') continue;
            if (largest_num < ch - '0') {
                result = largest_num;
                largest_num = ch - '0';
            } else if(largest_num > ch - '0' && result < ch - '0') {
                result = ch - '0';
            }
        }
        
        return result;
    }
};

File: Kotlin/1796_Second_Largest_Digit_in_a_String.kt
class Solution {
    fun secondHighest(s: String): Int {
        var result = -1
        var largestNum = -1
        
        for(ch in s) {
            if(!ch.isDigit()) continue
            if(largestNum < ch - '0') {
                result = largestNum
                largestNum = ch - '0'
            } else if(largestNum > ch - '0' && result < ch - '0') {
                result = ch - '0'
            }
        }
        
        return result
    }
}